### PR TITLE
Draft: fix Autogroup to fix Draft_OrthoArray crash when using active Std_Part.

### PR DESCRIPTION
**Pull request by carlopav**
Reference to https://forum.freecadweb.org/viewtopic.php?f=23&t=59155&p=508132#p508132

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/Draft/draftutils/gui_utils.py [hunk 1]
```diff
@@ -129,13 +129,19 @@ def autogroup(obj):
         if Gui.ActiveDocument.ActiveView.getActiveObject("Arch"):
             # add object to active Arch Container
             active_arch_obj = Gui.ActiveDocument.ActiveView.getActiveObject("Arch")
+            if obj in active_arch_obj.InListRecursive:
+                # Prevents cyclic references and avoid autogroup in obj points to active_arch_obj
+                return
             active_arch_obj.addObject(obj)
 
         elif Gui.ActiveDocument.ActiveView.getActiveObject("part", False) is not None:
```
Comment (Roy-043) [inline src/Mod/Draft/draftutils/gui_utils.py:137]: I think `is not None` can be removed.

**Comment by Roy-043** [comment]
Thanks for the fix.

**Comment by Roy-043** [comment]
There is a typo in the comments:
`# Prevents cyclic references and avoid autogroup in obj points to active_arch_obj`
Maybe change to:
`# do not autogroup if obj points to active_arch_obj to prevent cyclic references`

File: src/Mod/Draft/draftutils/gui_utils.py (hunk 1 above)
Comment (carlopav) [reply src/Mod/Draft/draftutils/gui_utils.py:137]: I'm not sure... being it beyond the scope of this PR i'd prefer to leave it like this.

File: src/Mod/Draft/draftutils/gui_utils.py [hunk 2]
```diff
@@ -128,13 +128,14 @@ def autogroup(obj):
...
-            Gui.ActiveDocument.ActiveView.getActiveObject("Arch").addObject(obj)
+            active_arch_obj = Gui.ActiveDocument.ActiveView.getActiveObject("Arch")
+            active_arch_obj.addObject(obj)
 
         elif Gui.ActiveDocument.ActiveView.getActiveObject("part", False) is not None:
             # add object to active part and change it's placement accordingly
             # so object does not jump to different position, works with App::Link
             # if not scaled. Modified accordingly to realthunder suggestions
-            p, parent, sub = Gui.ActiveDocument.ActiveView.getActiveObject("part", False)
+            active_part, parent, sub = Gui.ActiveDocument.ActiveView.getActiveObject("part", False)
```
Comment (matthijskooijman) [inline src/Mod/Draft/draftutils/gui_utils.py:141]: Yay for more readable names ;-)

**Pull request merged**